### DISCUSSION TITLE
Velocity based on delta time is inconsistent with framerates

### DISCUSSION THREAD

**Discussion by Hex4dec** [raysan5/raylib#4153] [Q&A]
I have some movement code that applies friction to/decays X velocity (`player->velocity.x = expdy(player->velocity.x, 0, 20, delta);`, where `expdy(a, b, decay, dt)` is the exponential decay function, TL;DR smooth curve), adds to the velocity when the player presses keys (`IsKeyDown(KEY_LEFT) -> player->velocity.x -= 80;`), and adds it to the position scaled by delta (`player->pos = Vector2Add(player->pos, Vector2Scale(player->velocity, delta));`. 

This is done before using a drawing function such as `BeginDrawing()` or `BeginTextureMode()`, but still in the process loop. I have seperated functions for processing and drawing. Why is this still framerate-dependent when I'm simulating an unstable FPS with `SetTargetFPS()`?

It seems to work as expected when I add directly to `player->pos`, which is even stranger.

This is most probably a stupid question, and I'm missing something very obvious.

**Comment by casperbear** [comment]
The phrase "Why is this still framerate-dependent when I'm simulating an unstable FPS with SetTargetFPS()?" is probably pointing to a misunderstanding about SetTargetFPS. Please see an answer to a similar question here https://github.com/raysan5/raylib/discussions/4201#discussioncomment-10369577

**Comment by Hex4dec** [comment]
Sorry, forgot to answer this. I fixed it by squaring delta in some places, acceleration is weird.
